crates/noirc_evaluator/src/ssa/conditional.rs

```diff
@@ -374,11 +374,13 @@ impl DecisionTree {
             merged_ins = self.synchronize(ctx, &left_ins, &right_ins, left);
         }
         let mut modified = false;
+        // write the merged instructions to the block
         super::optimizations::cse_block(ctx, left, &mut merged_ins, &mut modified)?;
         if modified {
             // A second round is necessary when the synchronization optimizes function calls between the two branches.
             // In that case, the first cse updates the result instructions to the same call and then
             // the second cse can (and must) then simplify identical result instructions.
+            // We clear the list because we want to perform the cse on the block instructions
```

Comment:
I don't think this comment is clear enough on why we want to clear the list. It is obvious without the comment that we want to call cse_block. What the comment doesn't tell us is e.g. why merged_ins are cleared for the second call but not the first.

Reply:
I think the real fix would be to refactor cse_block to have a consistent behaviour instead of relying on the provided list being empty or not. However I think this is not related to this PR and should be done in a separate one.

Reply:
Can we mention that in the comment then? I didn't know that was the case at all before this reply

Reply:
Done